tests/test_transformation/test_arcsin_transformer.py

```diff
@@ -0,0 +1,42 @@
+import pytest
+from sklearn.exceptions import NotFittedError
+
+from feature_engine.transformation import ArcsinTransformer
+
+
+def test_fit_raises_error_if_na_in_df(df_na):
+    # test case 2: when dataset contains na, fit method
+    with pytest.raises(ValueError):
+        transformer = ArcsinTransformer()
+        transformer.fit(df_na)
+
+
+def test_transform_raises_error_if_na_in_df(df_vartypes, df_na):
+    # test case 3: when dataset contains na, transform method
+    with pytest.raises(ValueError):
+        transformer = ArcsinTransformer()
+        transformer.fit(df_vartypes)
+        transformer.transform(df_na[["Name", "City", "Age", "Marks", "dob"]])
+
+
+def test_error_if_df_contains_outside_range_value(df_vartypes):
+    # test error when data contains value outside range [-1, +1]
+    df_neg = df_vartypes.copy()
+    df_neg.loc[1, "Age"] = 2
+
+    # test case 4: when variable contains value outside range, fit
+    with pytest.raises(ValueError):
+        transformer = ArcsinTransformer()
+        transformer.fit(df_neg)
+
+    # test case 5: when variable contains value outside range, transform
+    with pytest.raises(ValueError):
+        transformer = ArcsinTransformer()
+        transformer.fit(df_vartypes)
```

Comment:
we need to take lines 34 and 35 outside the pytest.raises, to ensure that these are not the lines that raise the error, so something like this:

```
transformer = ArcsinTransformer()
transformer.fit(df_vartypes)
with pytest.raises(ValueError):
    transformer.transform(df_neg)
```

having said this, this transformation will fail because of Age, so when starting the transformer pass Marks to the variables argument.